__tests__/app.test.js

```diff
@@ -142,6 +134,89 @@ describe("nc_games_test", () => {
     });
   });
+  describe("/api/reviews/:review_id/comments", () => {
+    it("200: GET /api/reviews/:review_id/comments should return an array of comments", () => {
+      return request(app)
+        .get("/api/reviews/3/comments")
+        .expect(200)
+        .then(({ body }) => {
+          const { comments } = body;
+          expect(comments.length).toBe(3);
+          expect(comments).toBeInstanceOf(Array);
+          comments.forEach((comment) => {
+            expect(comment).toBeInstanceOf(Object);
+          });
+        });
+    });
+    it("200: GET /api/reviews/:review_id/comments should have an expected shape", () => {
+      return request(app)
+        .get("/api/reviews/2/comments")
+        .expect(200)
+        .then(({ body }) => {
+          const { comments } = body;
+          expect(comments.length).toBe(3);
+          comments.forEach((comment) => {
+            expect(comment).toHaveProperty("comment_id", expect.any(Number));
+            expect(comment).toHaveProperty("votes", expect.any(Number));
+            expect(comment).toHaveProperty("author", expect.any(String));
+            expect(comment).toHaveProperty("body", expect.any(String));
+            expect(comment).toHaveProperty("review_id", expect.any(Number));
+          });
+        });
+    });
+    it("200: GET /api/reviews/2/comments?sort_by=created_at should respond with comments sorted by most recent first", () => {
+      return request(app)
+        .get("/api/reviews/2/comments?sort_by=created_at")
+        .expect(200)
+        .then(({ body }) => {
+          const { comments } = body;
+          expect(comments).toBeSortedBy("created_at", {
+            descending: true,
+          });
+        });
+    });
+    it("400: Should respond Invalid Query when any other sorts are attempted", () => {
+      return request(app)
+        .get("/api/reviews/3/comments?sort_by=body")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Invalid Query");
+        });
+    });
+    it("400: should only accept asc/desc as order query", () => {
+      return request(app)
+        .get("/api/reviews/2/comments?order=hotdogs")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Invalid Query");
+        });
+    });
+    it("400: invalid id type provided, should provide bad request message", () => {
+      return request(app)
+        .get("/api/reviews/bananas/comments")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Invalid input");
+        });
+    });
+    it("400: non-existent id provided, should provide bad request message", () => {
+      return request(app)
+        .get("/api/reviews/1010101010101/comments")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Invalid input");
+        });
+    });
+    it("200: GET /api/reviews/:review_id/comments if user has no comments should return a message", () => {
```

Comment:
we're not getting comments by their user here we're getting them by their review id, if what you're actually testing here is a review that has no comments on it it should be a 200 but it should return an empty array, not an error message